### PR TITLE
Refactor Datastore import to restore drush Datastore import

### PR DESCRIPTION
**Pull request by rhabbachi**
Issue: CIVIC-6317 & CIVIC-6396

## Description
Due to some aggressive refactoring, the drush command to update a resource datastore from a provided file was broken. The change introduced a disparity between the Base abstract class and the implemented method naming, as well as the method name and arguments provided for the drush command implemented.

This PR re-introduce some order by:
1. Using the same naming schema for the Base abstract class, the implementation class and the methods called by the drush commands.
2. Add more checking and  error reporting tools.
3. Better way to setup FeedsSource object for background tasks.
4. Better file cleanup logic.

## QA steps
* Updating the resource datastore with Drush (`drush dsu` command) should be restored.

### CODE REVIEW

File: modules/dkan/dkan_datastore/includes/DkanDatastore.inc [hunk 1]
```diff
@@ -523,40 +565,78 @@ class DkanDatastore extends Datastore implements DatastoreFormInterface {
...
-    $feed = isset($this->feed) ? $this->feed : $this->feed($this->node);
+   *
+   * @param FeedsSource $source
+   *   The FeedsSource object to work on.
+   *
+   * @return bool
+   *   TRUE if the new configuration is applyed to the source. FALSE if not.
+   */
+  protected function setupSourceBackground(FeedsSource &$source) {
+    var_dump($source instanceof FeedsSource);
```
Comment (dafeder) [inline modules/dkan/dkan_datastore/includes/DkanDatastore.inc]: Remove debug

File: modules/dkan/dkan_datastore/modules/dkan_datastore_api/dkan_datastore_api.drush.inc [hunk 2]
```diff
@@ -110,23 +110,44 @@ function dkan_datastore_api_datastore_rows_command($id = NULL) {
 /**
  * Callback for the datastore-update command.
  */
-function dkan_datastore_api_datastore_update_command($id = NULL, $file = NULL) {
+function dkan_datastore_api_datastore_update_command($id = NULL, $file_path = NULL) {
+  // Sanity check the arguments.
+  foreach (array('id', 'file_path') as $argument) {
+    if (!$$argument) {
+      drush_log(t('%argument is required.', array('%argument' => $argument)),
```
Comment (dafeder) [inline modules/dkan/dkan_datastore/modules/dkan_datastore_api/dkan_datastore_api.drush.inc]: Replace with drush_print

File: modules/dkan/dkan_datastore/modules/dkan_datastore_api/dkan_datastore_api.module [hunk 3]
```diff
@@ -599,49 +605,53 @@ function dkan_datastore_api_group_by(&$data_select, $group_by, $alias = DKAN_DAT
 function dkan_datastore_api_aggregations(&$data_select, $aggregations, $alias = NULL) {
 
   // Walk all the aggregations (sum, avg, etc.).
-  // { "sum": {"table_name": {"field_name": "as_field" }}}
+  // { "sum": {"table_name": {"field_name": "as_field" }}}.
```
Comment (dafeder) [inline modules/dkan/dkan_datastore/modules/dkan_datastore_api/dkan_datastore_api.module]: Remove any commented-out code that has no annotation

**Comment by rhabbachi** [comment]
Tests are failing because http://demo.getdkan.com/ is down.

**Comment by msolv** [comment]
Tests were passing before the Changelog was updated. Merging this without waiting for tests to pass again.

**Pull request merged**